Source/Core/Core/HW/EXI/BBA/TAPServer.cpp

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/HW/EXI/EXI_DeviceEthernet.h"
+
+#ifdef _WIN32
+#include <winsock2.h>
+#include <ws2ipdef.h>
+#else
+#include <fcntl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+#endif
+
+#include "Common/CommonFuncs.h"
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Core/HW/EXI/EXI_Device.h"
+
+namespace ExpansionInterface
+{
+
+#ifdef _WIN32
+static constexpr auto pi_close = &closesocket;
+using ws_ssize_t = int;
+#else
+static constexpr auto pi_close = &close;
+using ws_ssize_t = ssize_t;
+#endif
+
+#ifdef __LINUX__
+#define SEND_FLAGS MSG_NOSIGNAL
+#else
+#define SEND_FLAGS 0
+#endif
+
+static int ConnectToDestination(const std::string& destination)
+{
+  if (destination.empty())
+  {
+    ERROR_LOG_FMT(SP1, "Cannot connect: destination is empty\n");
+    return -1;
+  }
+
+  int ss_size;
+  struct sockaddr_storage ss;
+  memset(&ss, 0, sizeof(ss));
+  if (destination[0] != '/')
+  {
+    // IP address or hostname
+    size_t colon_offset = destination.find(':');
+    if (colon_offset == std::string::npos)
+    {
+      ERROR_LOG_FMT(SP1, "Destination IP address does not include port\n");
+      return -1;
+    }
+
+    struct sockaddr_in* sin = reinterpret_cast<struct sockaddr_in*>(&ss);
+    sin->sin_addr.s_addr = htonl(sf::IpAddress(destination.substr(0, colon_offset)).toInteger());
+    sin->sin_family = AF_INET;
+    sin->sin_port = htons(stoul(destination.substr(colon_offset + 1)));
+    ss_size = sizeof(*sin);
+#ifndef _WIN32
+  }
+  else
+  {
+    // UNIX socket
+    struct sockaddr_un* sun = reinterpret_cast<struct sockaddr_un*>(&ss);
+    if (destination.size() + 1 > sizeof(sun->sun_path))
```

Comment:
`destination.size() + 1 > sizeof(sun->sun_path)` is equivalent to `destination.size() >= sizeof(sun->sun_path)`. I consider the former to be clearer because the +1 explicitly accounts for the null terminator.